### DISCUSSION TITLE
LEDs not turning off (re: no source left and disabling the instance)

### DISCUSSION THREAD

**Discussion by sofakng** [awawa-dev/HyperHDR#123] [General]
I'm still new to HyperHDR but I'm having an issue with my SK6812 RGBW LEDs not turning off.

In the log I can see it disabling the LEDs:
`2021-09-20T16:53:44.590Z [HYPERHDR0] No source left -> switch LED-Device off`

...and I've also completed disabled the instance:
`2021-09-20T16:53:45.281Z [COMPONENTREG0] LED device: disabled` (not sure if that is the correct log message)

However, the LEDs are still illuminated:
![IMG_3499](https://user-images.githubusercontent.com/2228499/134043043-2e409545-e91c-44cb-94a5-2c1c1ab6cb7b.jpg)

I'm using HyperHDR v17 and HyperSerialESP32 (custom uploaded because I needed to change the data pin).

**Comment by awawa-dev** [comment]
Best way would be uploading your back up of configuration(general tab) & full logs (after the start when you try to disable the LED) then I will try to reproduce it on my setup. You can also try to disable smoothing/antiflickering filter/minimum backlit in the processing tab. Always that happens? Because AWA protocol can drop the 'black' command if the incoming frame is damaged.

**Reply by sofakng** [reply]
How can I export the full log?  Just copy and paste from the Log window?

**Reply by awawa-dev** [reply]
You can for example: copy them from Log window at put the content on pastebin service

**Reply by sofakng** [reply]
OK... I've rebooted the Raspberry Pi with HyperHDR.  I turned on my TV/AVR, waited 30 seconds (LEDs worked), and then turned it off.  However, the log doesn't seem to indicate the video has stopped?  The Live Video does show a black screen:

![image](https://user-images.githubusercontent.com/2228499/134048689-5dacccff-f31c-4fda-bde3-425bfced13ac.png)

Here is the log:  https://pastebin.com/WmGWd4uk

Afterwards, I went to [General] -> Disable Instance.

Here is the log:  https://pastebin.com/YQQChg8V

Smoothing is disabled.  I'll test with HyperSerialWLED and report back too.

**Comment by awawa-dev** [comment]
Also since it's modified HyperSerial please flash HyperSerialWLED (you can redefine pins in WLED) to confirm that LED driver is OK and the problem still persist.

**Reply by sofakng** [reply]
I'll give this a test.  Is there any benefit to using 2M instead of 921600 baud with 247 LEDs?

**Reply by awawa-dev** [reply]
Yes, for ~250leds and RGBW definitely go with 2Mb. WLED could wait for now.

OK, there are too many things configured in your setup and the grabber could lead to some misunderstanding because of producing 'almost' black background. Disable it completely for the test and use effect for example 'rainbow' effect and then disable LED device in the remote tab. Need backup of configuration.
Remember also about it:
> You can also try to disable smoothing/anti-flickering filter/minimum backlit in the processing tab

**Reply by awawa-dev** [reply]
Don't use 'disable instance' for now: remote tab -> turn off LED device component instead

**Reply by awawa-dev** [reply]
BTW still you have minimum backlit enabled: 2021-09-20T17:33:57.883Z [RGB_TRANSFORM0] setBacklightThreshold: 1

**Reply by sofakng** [reply]
OK - Here is my exported configuration:

[HyperHDR-17.0.0.0-Backup-2021-09-20.zip](https://github.com/awawa-dev/HyperHDR/files/7197838/HyperHDR-17.0.0.0-Backup-2021-09-20.zip)


I'm using all defaults from a fresh installation as far as I'm aware.  I tried to use the settings as you described and turning the TV on/off doesn't turn off the LEDs.  However, with all of the settings now if I use [Remote Control] -> [LED Device] (off), it does turn the LEDs off.

Here are the logs:  https://pastebin.com/Ku4n2VPE

**Reply by awawa-dev** [reply]
Thanks! But why turning the TV on/off should disable LEDs with your configuration?

**Reply by sofakng** [reply]
I thought when the HDMI signal is black (ie. tv/device is off) that it would turn off the LEDs?

**Reply by awawa-dev** [reply]
No, it does not work that way. For Rpi 2,3,4 there two (well... almost three 😉  ) ways but they need to be configured first:
1 ) using CEC, if Rpi is connected to the TV and TV is turned off/on it sends power off/on signal that could be received by Rpi if nothing blocks it on the way. But first: https://i.postimg.cc/NfM33H8p/control.jpg
2/3) using automatic or manual signal detection. Still both need to be configured first in the grabber properties. About new automatic signal detection you can read in the latest post on the hyperhdr blog.

**Reply by awawa-dev** [reply]
But if your are SURE that your grabber is sending TRUE black color (rarely happens, it is usually very dark gray) then PLEASE disable that minimal backlit :wink: Effect would be almost the same but the LED strip will still receive refresh command: signal detection disables LED completely.

**Reply by sofakng** [reply]
Ahhh, OK.  Thanks so much for the information and I apologize about the misunderstanding.

What setting are you talking about when you say to disable the minimal backlight?

**Reply by awawa-dev** [reply]
https://postimg.cc/Bj1zkHPN
set 'backlight treshold' to 0
but I doubt it will help in your case: you post the screenshot [link](https://user-images.githubusercontent.com/2228499/134048689-5dacccff-f31c-4fda-bde3-425bfced13ac.png) and you may think it's black but it's RGB(7,7,7) (if it's not disturbed by image's compression) rather far from black RGB(0,0,0) ..and your LED strip still could render some gray, not exactly the some values in short but it's not black.

If that's the case 'the signal detection' could be the only way.
